### ISSUE TITLE
no channels showing up since Kodi 17.5 on FireTV

### ISSUE THREAD

**Issue by cipher2k** [manuelm/pvr.dvbviewer#18]
I have been using DVBViewer add-on with Kodi 17.4 without any issues.
But, after updating to 17.5.1 no channels are showing up. EPG loading and recordings seems to work.

Downgraded to 17.4 and channel list shows up.

Also tried Kodi 18 - same result as 17.5.1: no channels.

**Comment by manuelm** [comment]
Please paste a link to a full debug log. However I'm wondering how you know EPG loading works if there are no channels.

**Comment by cipher2k** [comment]
not sure if it was EPG or something else, but I have seen in the status message that it was scanning some info from the dvb viewer service and it looked like some channel names appeared. it was very brief though and no channels appeared.

I will try to pull a debug log later.

**Comment by cipher2k** [comment]
I did a clean install of 17.5.1 and just added and configured the DVBViewer add-on. no success.

Here is the kodi.log: [http://ciesnik.de/kodi.log](url)

And here is the getchannelsxml: [http://ciesnik.de/getchannelsxml.xml](url)

**Comment by manuelm** [comment]
404 for both urls

**Comment by cipher2k** [comment]
yes, sorry. I corrected the URLs above.

Here is the kodi.log: [http://ciesnik.de/kodilog.txt](http://ciesnik.de/kodilog.txt)

And here is the getchannelsxml: [http://ciesnik.de/getchannelsxml.txt](http://ciesnik.de/getchannelsxml.txt)

**Comment by manuelm** [comment]
22:30:11.694 T:18446744072296596792   DEBUG: CurlFile::Open(0xabc651d8) http://192.168.90.3:10101/api/getchannelsxml.html?fav=1&subchannels=1&logo=1
22:30:11.706 T:18446744072296596792    INFO: AddOnLog: DVBViewer Client: Loaded (0/0) channels in (0/0) groups
22

mh `fav=1` looks wrong. You're using any favourites:
`22:30:11.668 T:18446744072295184520   DEBUG: AddOnLog: DVBViewer Client: Use favourites: no`

**Comment by manuelm** [comment]
err no, the PVR is always fetching the favourites and ignores them if disabled.

Can you fetch the above url and paste it somewhere please?

**Comment by cipher2k** [comment]
I did fetch it already. It is the second link.

**Comment by manuelm** [comment]
ah thanks. looks like your favourites and channels are merged in one group.

yours are like:
`<root>fav1,fav2,...,ch1,ch2,...</root>`

where as mine are:
`<root>fav1,fav2,...</root><root>ch1,ch2,...</root>`

Have to check the docs again.

**Comment by manuelm** [comment]
```
    fav=1 

    lets the API enumerate the favorites at the top of the channel list, also apply to “rootsonly” and “groupsonly” mode, which means, the favorite groups resp. the favorite root are enumerated 
```

I'll check with Griga. Merging those groups seems wrong imho.

**Comment by cipher2k** [comment]
Ok, in the meantime I have deleted all my favorites (I did not use them anyway). Now, the channels are being shown in Kodi 17.5.1

**Comment by manuelm** [comment]
Yeah, that works. The PVR is simply skipping the first group if favourites are disabled and the first group contains favourites (first channel number is negative). So in your merged group case all channels are skipped.

**Comment by manuelm** [comment]
note to myself: DMS 2.0.4.0

**Comment by manuelm** [comment]
So we (the dvbviewer team and myself) have discussed this internally and it looks like a bug in the DMS. However nothing that can easily tracked by reading the source code. So if you can reproduce this behavior again Griga can take a look. In particular he requests generating a support.zip, channels.dat,  ChannelsURL.ini (if created) and your favourites.xml.

**Comment by cipher2k** [comment]
Ok, you can download all requested files with this zip: [http://ciesnik.de/dvbv.zip](http://ciesnik.de/dvbv.zip)

**Comment by manuelm** [comment]
After more discussion Griga said this is expected behavior. You have named your main channel group `Favoriten` and favorites are hardcoded to be named `Favoriten` as well. DMS is merging channel those.

I haven't decided if I'll actually fix this bug.

**Comment by manuelm** [comment]
After even more discussion the next version of DMS will always create an independent channel directory for favourites, regardless of its name. I'll leave this issue open until the version has been released.
